engine/src/distribution/MessageManager.cpp

```diff
@@ -0,0 +1,48 @@
+#include "MessageManager.h"
```

Comment:
Now instead of having the logic of  managing the waiting and sending of these messages in a common place it has been split up. This is not a sound pattern its too easy for things to get out of whack and its not clear.

Reply:
can you explain what you mean

Reply:
I mean the logic that does send messages and wait for those messages to count are very related and should be done in the smae logical place. Here the generation of the message and the actual waiting for its other parts to arrive is now seperated